garak/resumeservice.py

```diff
@@ -0,0 +1,543 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Resume service for continuing interrupted garak scans.
+
+This service handles loading and managing resume state from checkpoint files,
+allowing scans to be resumed from where they left off.
+"""
+
+import hashlib
+import json
+import logging
+import os
+import uuid
+from collections import defaultdict
+from typing import Dict, Optional, Set, Tuple
+
+from garak import _config
+from garak.exception import GarakException
+
+
+# Module-level state (replaces _config.transient storage)
+_resume_state: Optional[Dict] = None
+
+
+class ResumeValidationError(GarakException):
+    """Raised when checkpoint validation fails."""
+
+    pass
```

Comment:
Exception definitions should be shared via the `garak.exception` package so users do not have to import every package to be able to handle errors.